cmd/controller-router.go

```diff
@@ -0,0 +1,42 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"net/rpc"
+
+	router "github.com/gorilla/mux"
+)
+
+// Routes paths for "minio control" commands.
+const (
+	controlPath = "/control"
+)
+
+// Register control RPC handlers.
+func registerControlRPCRouter(mux *router.Router, ctrlHandlers *controllerAPIHandlers) {
+	ctrlRPCServer := rpc.NewServer()
+	ctrlRPCServer.RegisterName("Control", ctrlHandlers)
+
+	ctrlRouter := mux.NewRoute().PathPrefix(reservedBucket).Subrouter()
+	ctrlRouter.Path(controlPath).Handler(ctrlRPCServer)
+}
+
+// Handler for object healing.
+type controllerAPIHandlers struct {
```

Comment:
the name can be `controlAPIHandler` as we use the term `control` at other places and not `controller`